src/fs/dir_builder.rs

```diff
@@ -36,7 +36,8 @@ impl DirBuilder {
     ///
     /// let builder = DirBuilder::new();
     /// ```
-    pub fn new() -> DirBuilder {
+    #[must_use]
+    pub const fn new() -> DirBuilder {
```

Comment:
What is the benefit of making this const?

Reply:
The benefit of making this const is that `DirBuilder::new` can now be used in const declarations, as well as anything that depends on it also now has a chance to be made const.